src/v/redpanda/admin/api-doc/partition.json

```diff
@@ -215,6 +215,21 @@
                     ]
                 }
             ]
+        },
+        {
+            "path": "/v1/partitions/leaders",
```

Comment:
IMO the api `DELETE "/v1/partitions/leaders` is misleading because leaders are a real entity externally visible to the system, while this API is specially altering transient cache.

What do we think about creating either (1) an api specifically for the metadata cache resource or (2) an extensible API for handling this and other types of data reset operations?

Reply:
This is such a debug-ish thing, I wonder if we should put is somewhere other than the main /v1/partitions path.

I don't know the right naming, but something like /v1/debug/reset_leaders or something like that.  I'm sort of anticipating any future "buttons" we want to add here, in a place where it'll be reasonably obvious to the user that this is something arcane and not a "normal" API.

Reply:
> This is such a debug-ish thing, I wonder if we should put is somewhere other than the main /v1/partitions path.

totally. I think I meant to say that explicitly

Reply:
@mmaslankaprv Could you please take a look.

I think it is good idea to move this request to /v1/debug section.
Should it be POST request with name reset_leaders? Or DELETE request with name leaders/leaders_info?